zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/ZoneService.java

```diff
@@ -1,23 +1,107 @@
+import org.cloudfoundry.identity.uaa.scim.ScimGroupProvisioning;
+import org.cloudfoundry.identity.uaa.constants.OriginKeys;
+import org.cloudfoundry.identity.uaa.scim.ScimGroup;
+import org.cloudfoundry.identity.uaa.saml.SamlKey;
+import org.cloudfoundry.identity.uaa.util.JsonUtils;
+import org.cloudfoundry.identity.uaa.resources.QueryableResourceManager;
+import org.cloudfoundry.identity.uaa.client.ClientDetailsValidator.Mode;
+import org.cloudfoundry.identity.uaa.client.ClientDetailsValidator;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.common.util.RandomValueStringGenerator;
+import org.springframework.security.oauth2.provider.client.BaseClientDetails;
+import org.springframework.security.oauth2.provider.ClientAlreadyExistsException;
+import org.springframework.security.oauth2.provider.ClientDetails;
+import org.springframework.security.oauth2.provider.NoSuchClientException;
+
+import org.springframework.util.StringUtils;
 
 public class ZoneService {
 
-    public static final String X_IDENTITY_ZONE_ID = "X-Identity-Zone-Id";
+    private static final Logger logger = LoggerFactory.getLogger(ZoneService.class);
 
+    public static final String X_IDENTITY_ZONE_ID = "X-Identity-Zone-Id";
+    public static final String GENERATED_KEY_ID = "generated-saml-key";
+    private static final java.util.Base64.Encoder base64encoder = java.util.Base64.getMimeEncoder(64, "\n".getBytes());
     private final IdentityZoneProvisioning zoneProvisioning;
+    private final IdentityProviderProvisioning idpProvisioning;
+    private final ScimGroupProvisioning groupProvisioning;
+    private final MultitenantClientServices multitenantClientServices;
+    private final QueryableResourceManager<ClientDetails> clientDetailsService;
+    private final ClientDetailsValidator clientDetailsValidator;
     private final String uaaDashboardUri;
+    private final String clientId;
+    private final String zoneAuthorities;
+    private final String grantTypes;
+    private final String resourceIds;
+    private final String scopes;
+    private final String uaaClientID;
+    private final String domainName;
 
-    public ZoneService(IdentityZoneProvisioning zoneProvisioning, String uaaDashboardUri) {
+    public ZoneService(IdentityZoneProvisioning zoneProvisioning,
+                       IdentityProviderProvisioning idpProvisioning,
+                       ScimGroupProvisioning groupProvisioning,
+                       MultitenantClientServices multitenantClientServices,
+                       QueryableResourceManager<ClientDetails> clientDetailsService,
+                       ClientDetailsValidator clientDetailsValidator,
+                       String uaaDashboardUri, String clientId, String zoneAuthorities, String grantTypes,
+                       String resourceIds, String scopes, String uaaClientID, String uaaUrl
```

Comment:
Number of constructor parameters looks excessive.  How come neither service broker [UaaZoneProvisioner](https://github.build.ge.com/predix/security-service-broker/blob/develop/service-broker-service/src/main/java/com/ge/predix/servicebroker/service/UaaZoneProvisioner.java#L73) nor UAA [JdbcIdentityZoneProvisioning](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/server/src/main/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioning.java#L49) required that many parameters?

Reply:
Necessary, qualifiers and required parameters have been kept as part of constructer inject of service.